### ISSUE TITLE
Report missing spaces on function annotations with defaults

### ISSUE THREAD

**Issue by webjunkie** [PyCQA/pycodestyle#692]
Following up on #357, I want to highlight that [PEP8 recommends](https://www.python.org/dev/peps/pep-0008/#other-recommendations) using spaces around the = sign for function annotations with a default:

> When combining an argument annotation with a default value, use spaces around the = sign (but only for those arguments that have both an annotation and a default).
> 
> Yes:
> 
>     def munge(sep: AnyStr = None): ...
>     def munge(input: AnyStr, sep: AnyStr = None, limit=1000): ...
> 
> No:
> 
>     def munge(input: AnyStr=None): ...
>     def munge(input: AnyStr, limit = 1000): ...

Why does pycodestyle not report this as a problem when done incorrectly?

Example:

```python
def foo(bar: str = False):
    pass


def foo_2(bar: str=False):
    pass
```

This is also a problem for autopep8, because when not reported by pycodestyle, it makes it harder for autopep8 to fix. See hhatto/autopep8#340

**Comment by AndrewUshakov** [comment]
Is it possible to "allow" variant: `def a(bar:int, foo:bool=False):` without disabling E231?

**Comment by taion** [comment]
#717 adds this as a new E252.
